copy-to-gcloud.sh

```diff
@@ -24,40 +24,77 @@ echo "LOCAL_DIRNAME=$LOCAL_DIRNAME"
 echo "GCLOUD_PATH=$GCLOUD_PATH"
 
 
+function copy_path_if_exists {
+    # Two arguments
+    # 1) local path to file
+    # 2) remote name of file relative to GCLOUD_PATH
+    SOURCE="$1"
+    DEST="gs://$GCLOUD_PATH/$2"
+    if [[ -f "$SOURCE" ]]; then
+        gsutil -m cp "$SOURCE" "$DEST"
+     else
+        echo "Skipping $SOURCE, file not found"
+    fi
+}
+
+function copy_local_file_if_exists {
+    # Prepends $LOCAL_DIRNAME to first argument and then
+    # calls copy_if_exists
+    #
+    # Two arguments:
+    # 1) local name of file relative to LOCAL_DIRNAME
+    # 2) remote name of file relative to GCLOUD_PATH
+    SOURCE="$LOCAL_DIRNAME/$1"
+    DEST="$2"
+    copy_path_if_exists "$SOURCE" "$DEST"
+}
 
-echo "=== Copying FastQC files ===" 
-if [ -d $LOCAL_DIRNAME/fastqc-output/ ]; then 
-  gsutil -m cp -r $LOCAL_DIRNAME/fastqc-output/ gs://$GCLOUD_PATH/
-else
-  echo "Skipping $LOCAL_DIRNAME/fastqc-output, directory not found"
-fi 
+function copy_directory_if_exists {
+    # One argument:
+    #   1) Local sub-directory relative to LOCAL_DIRNAME
+    # Will be copied into GCLOUD_PATH
+    SOURCE=$LOCAL_DIRNAME/$1
+    # remove trailing slashes to avoid double slashes
+    SOURCE=${SOURCE%/}
+    DEST="gs://$GCLOUD_PATH/"
+    if [[ -d FULL_DIRPATH ]]; then
+        gsutil -m cp -r "$SOURCE/" "$DEST"
+     else
+        echo "Skipping $SOURCE, directory not found"
+    fi
+}
+
+echo "=== Copying FastQC files ==="
+copy_directory_if_exists "fastqc-output"
 
 echo "=== Copying Picard metrics ==="
-gsutil -m cp -R $LOCAL_DIRNAME/*metrics*.txt gs://$GCLOUD_PATH/picard-metrics/
+for f in "$LOCAL_DIRNAME"/*metrics*.txt; do
+    copy_path_if_exists "$f" picard-metrics/`basename "$f"`
+done
 
 echo "=== Copying logs ==="
-gsutil -m cp -R $LOCAL_DIRNAME/logs/ gs://$GCLOUD_PATH/
+copy_directory_if_exists "logs"
 
 echo "=== Copying VCF files ==="
 for vcf in mutect.vcf mutect2.vcf strelka.vcf filtered_normal_germline_snps_indels.vcf filtered_covered_normal_germline_snps_indels.vcf
 do
-    if [ -f $LOCAL_DIRNAME/$vcf ]
-    then
-        gsutil -m cp $LOCAL_DIRNAME/$vcf gs://$GCLOUD_PATH/$vcf
-    else   
-        echo "Skipping $LOCAL_DIRNAME/$vcf, does not exist"
-    fi
+    copy_local_file_if_exists "$vcf" "$vcf"
 done
 
 echo "=== Copying Vaxrank output ==="
-gsutil -m cp $LOCAL_DIRNAME/vaccine-peptide-report* gs://$GCLOUD_PATH/
-gsutil -m cp $LOCAL_DIRNAME/all-passing-variants*.csv gs://$GCLOUD_PATH/
+for f in "$LOCAL_DIRNAME"/vaccine-peptide-report*; do
+    copy_path_if_exists "$f" `basename "$f"`
+done
+for f in "$LOCAL_DIRNAME"/all-passing-variants*.csv; do
+    copy_local_file_if_exists "$f" `basename "$f"`
```

Comment:
a little confused here - why is this `copy_local_file_if_exists` but the loop above it is using `copy_path_if_exists`?